mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java

```diff
@@ -231,4 +242,81 @@ public void searchShouldNotReturnResultsFromOtherUsers() throws Exception {
         assertThat(metaDatas.get(0).getPath()).isEqualTo(MailboxPath.inbox(session));
     }
 
+    @ContractTest
+    public void updateAnnotationsShouldThrowExceptionWithInvalidKey() throws BadCredentialsException, MailboxException, InvalidAnnotationException {
+        expected.expect(InvalidAnnotationException.class);
+        session = mailboxManager.createSystemSession(USER_2, LoggerFactory.getLogger("Test"));
+        MailboxPath inbox = MailboxPath.inbox(session);
+        mailboxManager.createMailbox(inbox, session);
+
+        mailboxManager.updateAnnotations(inbox, session, Lists.newArrayList(INVALID_KEY_ANNOTATION));
+    }
+
+    @ContractTest
+    public void updateAnnotationsShouldThrowExceptionWithNullKey() throws BadCredentialsException, MailboxException, InvalidAnnotationException {
+        expected.expect(NullPointerException.class);
+        session = mailboxManager.createSystemSession(USER_2, LoggerFactory.getLogger("Test"));
+        MailboxPath inbox = MailboxPath.inbox(session);
+        mailboxManager.createMailbox(inbox, session);
+
+        mailboxManager.updateAnnotations(inbox, session, Lists.newArrayList(MailboxAnnotation.newInstance(null, "value")));
+    }
+
+    @ContractTest
+    public void updateAnnotationsShouldUpdateStoredAnnotation() throws BadCredentialsException, MailboxException, InvalidAnnotationException {
```

Comment:
Thanks, I will bear in mind.
So basically the setup on our CI has problem. It does not run all testing